src/inference/src/cpp_interfaces/interface/ie_iplugin_internal.cpp

```diff
@@ -323,7 +323,26 @@ void SetExeNetworkInfo(const std::shared_ptr<IExecutableNetworkInternal>& exeNet
     const auto& inputsInfo = exeNetwork->GetInputsInfo();
     const auto& outputsInfo = exeNetwork->GetOutputsInfo();
     OPENVINO_ASSERT(inputsInfo.size() == function->get_parameters().size());
-    OPENVINO_ASSERT(outputsInfo.size() == function->get_output_size());
+
+    if (outputsInfo.size() != function->get_output_size()) {
+        std::unordered_set<std::string> output_names;
+        size_t duplicates_count = 0;
+        for (const auto& result : function->get_results()) {
+            const auto& name = result->input_value(0).get_any_name();
```

Comment:
Can we just check the number of unique output tensors from `function->outputs()`?

Reply:
I've changed the logic to use `function->outputs()` and `get_tensor_ptr()`. Thanks!